test/functional/p2p_invalid_messages.py

```diff
@@ -24,8 +27,7 @@
     wait_until,
 )
 
-MSG_LIMIT = 4 * 1000 * 1000  # 4MB, per MAX_PROTOCOL_MESSAGE_LENGTH
-VALID_DATA_LIMIT = MSG_LIMIT - 5  # Account for the 5-byte length prefix
+VALID_DATA_LIMIT = MAX_PROTOCOL_MESSAGE_LENGTH - 5  # Account for the 5-byte length prefix
```

Comment:
We set `VALID_DATA_LIMIT` so that the length of the serialised `msg_unrecognized` message built with it will equal `MAX_PROTOCOL_MESSAGE_LENGTH`. This is verified at line 153; you can also add the following assertion at line 111
```diff
             msg = msg_unrecognized(str_data="d" * (VALID_DATA_LIMIT + 1))
+            assert_equal(len(msg.serialize()), MAX_PROTOCOL_MESSAGE_LENGTH + 1)
```
or try changing the offset from 5 to 4 or 6.

I've added this assertion and some other ones while testing @troygiorshev's net message headers refactoring and may propose the changes if they look worthwhile to me after a few days' time.